### PR TITLE
refactor: Make explicit CMutableTransaction -> CTransaction conversion.

### PR DESCRIPTION
**Pull request by lucash-dev**
This PR is re-submission of #14156, which was automatically closed by github (glitch?)

Original description: 

This PR makes explicit the now implicit conversion constructor `CTransaction(const CMutableTransaction&)` in `transaction.h`. 
Minimal changes were made elsewhere to make the code compilable. I'll follow up with other PRs to address individually refactoring functions that should have a `CMutableTransaction` version, or where a `CTransaction` should be reused. 

The rationale for this change is:

 - Conversion constructors should not be explicit unless there's a strong reason for it (in the opinion of, for example, https://google.github.io/styleguide/cppguide.html, and https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ro-conversion. Let me know your take on this).
 - This particular conversion is very costly -- it implies a serialization plus hash of the transaction.
 - Even though `CTransaction` and `CMutableTransaction` represent the same data, they have very different use cases and performance properties.
 - Making it explicit allows for easier reasoning of performance trade-offs.
 - There has been previous performance issues caused by unneeded use of this implicit conversion.
 - This PR creates a map for places to look for possible refactoring and performance gains (this benefit still holds if the PR is not merged).

### CODE REVIEW

**Comment by lucash-dev** [comment]
The part referring to tests and benchmark changes has now a separate PR #14908

File: src/qt/bitcoin.cpp [hunk 1]
```diff
@@ -71,11 +71,6 @@ Q_DECLARE_METATYPE(bool*)
 Q_DECLARE_METATYPE(CAmount)
 Q_DECLARE_METATYPE(uint256)
 
-static void InitMessage(const std::string& message)
```
Comment (promag) [inline src/qt/bitcoin.cpp]: 👀

File: src/qt/bitcoin.cpp [hunk 2]
```diff
@@ -563,6 +558,11 @@ int main(int argc, char *argv[])
 
     std::unique_ptr<interfaces::Node> node = interfaces::MakeNode();
 
+    // Subscribe to global signals from core
```
Comment (promag) [inline src/qt/bitcoin.cpp]: 👀

**Review by promag — COMMENTED** [review]
Looks like you are missing a rebase.

File: src/qt/bitcoin.cpp (hunk 1 above)
Comment (lucash-dev) [reply src/qt/bitcoin.cpp]: Definitely some rebasing weirdness.

**Comment by lucash-dev** [comment]
Rebased

**Comment by gwillen** [comment]
Strong support for making constructors explicit wherever possible. This looks like a fairly minimal change in terms of total amount of stuff touched, and definitely in a positive direction. Would love to see this get in. (Consider this a utACK from me, for what my vote is worth. I'm not sure what's up with Travis but the failure doesn't look legitimate.)

**Comment by lucash-dev** [comment]
Rebased on latest master. Everything worked locally. Let's see how Travis behaves.

**Comment by MarcoFalke** [comment]
utACK b301950df32443e358bc22ca22c6f9ac09d18219

**Comment by practicalswift** [comment]
utACK b301950df32443e358bc22ca22c6f9ac09d18219

**Pull request merged**